src/vr/journal.zig

```diff
@@ -489,129 +729,241 @@ pub const Journal = struct {
         };
     }
 
-    /// Writes headers between `op_min` and `op_max` (both inclusive).
-    fn write_headers_between(self: *Journal, op_min: u64, op_max: u64) void {
-        // TODO Snapshots
-        assert(op_min <= op_max);
+    /// Attempt to lock the in-memory sector containing the header being written.
+    /// If the sector is already locked, add this write to the wait queue.
+    fn write_prepare_header(self: *Journal, write: *Journal.Write) void {
```

Comment:
Shall we shift this up to just before `write_prepare_on_write_header()`?

Reply:
You're right that the current order of this group of functions is pretty messy. I've reorganized them in a way that I think reads much better :+1: